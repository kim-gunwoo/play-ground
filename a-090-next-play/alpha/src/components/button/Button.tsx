import React from "react";
import styles from './Button.module.sass'

interface IButtonProps {
    children?: React.ReactNode;
    onClick?: React.MouseEventHandler<HTMLButtonElement>
}

export default function Button({ children, onClick }: IButtonProps) {
    return (
        <button
            className={styles.button}
            onClick={onClick}
        >
            {children}
        </button>
    )
}