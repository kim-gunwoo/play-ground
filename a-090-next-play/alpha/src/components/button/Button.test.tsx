import { fireEvent, render, screen } from "@testing-library/react";
import Button from "./Button";

describe('Button', () => {
    it('renders Button', () => {
        const onClick = jest.fn(); // 가짜 함수
        const { getByText } = render(<Button onClick={onClick} >버튼</Button>);
        const button = getByText('버튼');
        expect(button).toBeInTheDocument();
        fireEvent.click(button);
        expect(onClick).toBeCalledTimes(1);
    });

    it('snapshot Button', () => {
        const { container } = render(<Button>버튼</Button>);
        expect(container).toMatchSnapshot();
    })

    it('renders Button2', () => {
        const onClick = jest.fn(); // 가짜 함수
        render(<Button onClick={onClick} >버튼</Button>);
        const button = screen.getByRole('button', { name: '버튼' })
        expect(button).toBeInTheDocument();
        fireEvent.click(button);
        expect(onClick).toBeCalledTimes(1);
    });
});