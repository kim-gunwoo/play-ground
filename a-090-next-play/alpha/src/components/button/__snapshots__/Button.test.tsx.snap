// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot Button 1`] = `
<div>
  <button
    class="button"
  >
    버튼
  </button>
</div>
`;
